### DISCUSSION TITLE
Performance issue

### DISCUSSION THREAD

**Discussion by WittObk** [ElinamLLC/SharpVectors#215] [Q&A] [answered]
I have an application with more than 10000 (device) images ... if i use the Image control and png's the initialization of the view is really fast but if i use the SvgViewbox it is very slow ... is there a way to optimize that?

**Comment by paulushub** [comment, marked as answer]
@obi111 By Image control do you mean with the `SvgImage` objects or displaying normal PNG images?

**Reply by WittObk** [reply]
Normal png images!

**Reply by WittObk** [reply]
Wpf Image class with Source property

**Reply by paulushub** [reply]
* First, the two processes in this case cannot be compared for speed. Using SVG file means you will first convert to a static image format similar to the PNG and then render it.
* Second, try using the WPF Image class with the Source property set to **SvgImage** to see if there is an improvement.
* Third, load the images yourself using the **FileSvgReader** class. This can allow you to load the images in multiple threads (see the **WpfTestThreadSafety** sample). Freeze the images, since that could improve the speed.
This will depend on you application, but I do think you do not need to display all 10000 images at the same time. So, you can with the third approach dynamically load the images on demand as used in virtual list views to speed up and reduce memory.

**Reply by WittObk** [reply]
yes thanks that works good!

**Reply by paulushub** [reply]
@obi111 Thanks for the feedback. 
Which method did you choose? (just for the records for others)
